QuizApp/Scenes/QuizPage/ViewController/QuizViewController.swift

```diff
@@ -254,15 +247,98 @@ final class QuizViewController: UIViewController {
         )
     }
     
+    // MARK: - Show PopUps
+    private func showScorePopup() {
+        let popup = FinalScorePopUp()
+        popup.configure(points: viewModel.quizScore)
+        
+        popup.closeAction = {
+            popup.removeFromSuperview()
+        }
+        
+        if let window = UIApplication.shared.windows.first(where: { $0.isKeyWindow }) {
+            popup.frame = window.bounds
+            window.addSubview(popup)
+            
+            popup.translatesAutoresizingMaskIntoConstraints = false
+            NSLayoutConstraint.activate(
+                [
+                    popup.leadingAnchor.constraint(
+                        equalTo: window.leadingAnchor,
+                        constant: Constants.Sizing.popupSidePadding
+                    ),
+                    popup.trailingAnchor.constraint(
+                        equalTo: window.trailingAnchor,
+                        constant: -Constants.Sizing.popupSidePadding
+                    ),
+                    popup.topAnchor.constraint(
+                        equalTo: window.topAnchor,
+                        constant: Constants.Sizing.popupVerticalPadding
+                    ),
+                    popup.bottomAnchor.constraint(
+                        equalTo: window.bottomAnchor,
+                        constant: -Constants.Sizing.popupVerticalPadding
+                    )
+                ]
+            )
+        }
+    }
+    
+    private func showClosePopup() {
+        let popup = CustomPopUp()
+        popup.configure(question: Constants.Texts.leaveQuizText)
+        
+        popup.onAcceptAction = { [weak self] in
+            popup.removeFromSuperview()
+            let homeViewModel = HomeViewModel()
+            let vc = HomeViewController(viewModel: homeViewModel)
+            self?.navigationController?.pushViewController(vc, animated: false)
+        }
+        
+        popup.onRejectAction = {
+            popup.removeFromSuperview()
+        }
+        
+        if let window = UIApplication.shared.windows.first(where: { $0.isKeyWindow }) {
```

Comment:
Check if you really need these lines because the ViewController is already in a NavigationController stack